test/spec/unit/core/adapterManager_spec.js

```diff
@@ -1142,5 +1144,85 @@ describe('adapterManager tests', function () {
         expect(bidRequests[0].gdprConsent).to.be.undefined;
       });
     });
+
+    describe('s2sTesting - testServerOnly', () => {
```

Comment:
I think there's a good mix of tests for requests who have `bidSource` set at the adUnit level, but in addition I think we need other test cases, for example:

1. a bidder whose requests always go server because they are in `s2sConfig.bidders` but are in neither `s2sConfig.bidderControl` nor has `bidSource` set at the adUnit level. Maybe just add a bidder like this to all test scenarios which should verify it does not affect the decision to suppress client requests.
2. no bidder has `bidSource` set at the adUnit level (only `s2sConfig.bidderControl`)
   a. 1 or more bidder set to server (should suppress all client bids)
   b. no bidders set to server (should not suppress any client bids)
3. combination of bidders with `bidSource` set at the adUnit level and other bidders set in `s2sConfig.bidderControl`
    a. nothing set to server (should not suppress any client bids)
    b. `bidderConrol` sets one bidder to server, `bidSource` does not (should suppress all client bids)
    c.  `bidSource` sets one bid to server, `bidderControl` does not (should suppress all client bids)
    d.  all set to server (should suppress all client bids)
4. no `bidSource` nor `bidderControl` bids are `server` (should not suppress any client bids)
5. `testServerOnly` is `false` or not defined when `bidSource` or `bidderConrol` has `server` bids (should not suppress any client bids)

Reply:
Skipping number five as it is a test already covered by previous `s2sTesting` unit tests.